File: 1Urte desberdinetako materiala/Lorea 25-26/Android/Erronka-1/multi_intefaz/src/main/java/model/Book.java

package model;


public class Book {
     private int id;
     private String izenburua;
     private int  idazlea_id;

    public Book(int id, String izenburua) {
        this.id = id;
        this.izenburua = izenburua;
    }

    public Book() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getIzenburua() {
        return izenburua;
    }

    public void setIzenburua(String izenburua) {
        this.izenburua = izenburua;
    }

    public int getIdazlea_id() {
        return idazlea_id;
    }

    public void setIdazlea_id(int idazlea_id) {
        this.idazlea_id = idazlea_id;
    }

    @Override
    public String toString() {
        return "Book{" + "id=" + id + ", izenburua=" + izenburua + ", idazlea_id=" + idazlea_id + '}';
    }
     
     
     
     
}

File: PMD 25-26/Erronka/app/src/main/java/com/example/erronka/model/Book.kt
package com.example.erronka.model

/**class Book {
    var id: Int = 0
    var izenburua: String? = null
    var idazlea_id: Int = 0

    constructor(id: Int, izenburua: String?, idazle: Int) {
        this.id = id
        this.izenburua = izenburua
        this.idazlea_id = idazle
    }

    constructor()

    override fun toString(): String {
        return "Book{" + "id=" + id + ", izenburua=" + izenburua + ", idazlea_id=" + idazlea_id + '}'
    }
}**/
class Book {

    private var id: Int = 0
    private var izenburua: String = ""
    private var idazlea_id: Int = 0

    // Constructor con parámetros
    constructor(id: Int, izenburua: String) {
        this.id = id
        this.izenburua = izenburua
    }

    // Constructor vacío
    constructor()

    // Getters y Setters
    fun getId(): Int {
        return id
    }

    fun setId(id: Int) {
        this.id = id
    }

    fun getIzenburua(): String {
        return izenburua
    }

    fun setIzenburua(izenburua: String) {
        this.izenburua = izenburua
    }

    fun getIdazlea_id(): Int {
        return idazlea_id
    }

    fun setIdazlea_id(idazlea_id: Int) {
        this.idazlea_id = idazlea_id
    }

    // toString
    override fun toString(): String {
        return "Book(id=$id, izenburua=$izenburua, idazlea_id=$idazlea_id)"
    }
}
